sharding/indexHashedNodesCoordinator.go

```diff
@@ -503,31 +519,73 @@ func (ihgs *indexHashedNodesCoordinator) EpochStartPrepare(metaHeader data.Heade
 		log.Error("saving nodes coordinator config failed", "error", err.Error())
 	}
 
-	displayNodesConfiguration(eligibleMap, waitingMap, leaving, stillRemaining, nodesConfig.nbShards)
+	displayNodesConfiguration(eligibleMap, waitingMap, newNodesConfig.leavingList, stillRemaining, newNodesConfig.nbShards)
 
 	ihgs.mutSavedStateKey.Lock()
 	ihgs.savedStateKey = randomness
 	ihgs.mutSavedStateKey.Unlock()
 }
 
-// ComputeActuallyLeaving returns the list of those nodes which are actually leaving
-func ComputeActuallyLeaving(leaving []Validator, stillRemaining []Validator) []Validator {
-	actualLeaving := make([]Validator, 0)
-	for _, shouldLeave := range leaving {
-		willRemain := false
-		for _, remains := range stillRemaining {
-			if bytes.Equal(shouldLeave.PubKey(), remains.PubKey()) {
-				willRemain = true
-				break
-			}
+// GetChance will return default chance
+func (ihgs *indexHashedNodesCoordinator) GetChance(_ uint32) uint32 {
+	return defaultSelectionChances
+}
+
+func (ihgs *indexHashedNodesCoordinator) computeNodesConfigFromList(
+	validatorInfos []*state.ShardValidatorInfo,
+) (*epochNodesConfig, error) {
+
+	leaving, err := ihgs.nodesPerShardSetter.ComputeLeaving(validatorInfos)
```

Comment:
leaving should be sorted also, either here or in the ComputeLeaving

Reply:
done

Reply:
ok